great_expectations/expectations/metrics/table_metrics/table_head.py

```diff
@@ -91,14 +91,21 @@ def _sqlalchemy(
                         con=execution_engine.engine,
                     )
                 else:
-                    df = next(
-                        pd.read_sql_table(
-                            table_name=getattr(selectable, "name", None),
-                            schema=getattr(selectable, "schema", None),
-                            con=execution_engine.engine,
-                            chunksize=metric_value_kwargs["n_rows"],
-                        )
```

Comment:
This was a SQL Alchemy bug; when `n_rows` `chunksize` was negative the entire dataframe was returned instead of an `Iterator` of partitions.